### PR TITLE
Layout selector performance improvements

### PR DESCRIPTION
**Pull request by jrtashjian**
### Description
The way we were using the BlockPreview component from Gutenberg was very slow when loading more than 5 previews in a list. With the release of Block Patterns in Gutenberg I've recognized performance improvements that were made in core where they BlockPreview component was being used inth block inserter. This PR takes those findings and implements them here.

The result is a more snappy interaction when interacting with the previews and no delay in interacting with the layout selector while previews load.

### Types of changes
Enhancement

### How has this been tested?
This has been tested manually with 1, 5, 10, 20, 50, and 100 layouts being added through the available filters in PHP.

### Checklist:
- [x] My code is tested
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [x] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: includes/class-coblocks-block-assets.php [hunk 1]
```diff
@@ -295,7 +296,7 @@ public function frontend_scripts() {
 		$vendors_dir = CoBlocks()->asset_source( 'js', 'vendors' );
 
 		// Masonry block.
-		if ( has_block( 'coblocks/gallery-masonry' ) || has_block( 'core/block' ) ) {
+		if ( is_admin() || has_block( 'coblocks/gallery-masonry' ) || has_block( 'core/block' ) ) {
```
Comment (richtabor) [inline includes/class-coblocks-block-assets.php:299]: Is there a check for a Gutenberg admin page instead?

File: src/extensions/layout-selector/editor.scss [hunk 2]
```diff
@@ -4,36 +4,6 @@ $color-highlight: #00a4a6;
 	opacity: 1 !important;
 }
 
-.coblocks-layout-selector .coblocks-layout-selector__layout .block-editor-block-preview__container.editor-styles-wrapper {
```
Comment (richtabor) [inline src/extensions/layout-selector/editor.scss:7]: Why did you remove these styles?

File: src/extensions/layout-selector/index.php [hunk 3]
```diff
@@ -46,6 +50,128 @@ function coblocks_layout_selector_categories() {
 function coblocks_layout_selector_layouts() {
 	$layouts = array();
 
+	for ( $i=0; $i <= 100; $i++ ) {
```
Comment (richtabor) [inline src/extensions/layout-selector/index.php:53]: Wonder if we should have one for testing, if WP_DEBUG is on, or a test var we can state?

Comment (jrtashjian) [reply src/extensions/layout-selector/index.php]: This was mistakenly committed. I've removed it.

File: includes/class-coblocks-block-assets.php (hunk 1 above)
Comment (jrtashjian) [reply includes/class-coblocks-block-assets.php:320]: Yes, we actually have `is_page_gutenberg()` we can use. Switching to it in the next push.

File: src/extensions/layout-selector/editor.scss (hunk 2 above)
Comment (jrtashjian) [reply src/extensions/layout-selector/editor.scss]: Probably just did it to strip every possible override during experimentation. This will continue to exist.

**Comment by jonathanbardo** [comment]
@jrtashjian Can you take a look at the conflict? Pretty sure it's only because we took out image fetch.

File: src/extensions/layout-selector/layout-selector-results.js [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+				 *
+				 * @param {Array} blocks The block objects of the layout.
+				 */
+				blocks: applyFilters( 'coblocks.layoutPreviewBlocks', layout.blocks ),
+			} ) ),
+		[ layouts, category ]
+	);
+
+	// Needed to render our list of previews asynchronously for better performance.
+	const currentShowLayouts = useAsyncList( filteredLayouts );
```
Comment (jonathanbardo) [inline src/extensions/layout-selector/layout-selector-results.js:35]: Oh that's pretty cool. Nice find

File: src/extensions/layout-selector/layout-selector-results.js [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+/**
+ * Renders the layout's block preview.
+ */
+export const LayoutPreview = ( { layout, onClick } ) => {
+	const [ overlay, setOverlay ] = useState( false );
+	const isSelected = false;
+
+	return (
+		<a href="#!"
```
Comment (jonathanbardo) [inline src/extensions/layout-selector/layout-selector-results.js:67]: @jrtashjian liked we chat about, I would remove the anchor here and move to a button. Replace isSelected with overlay and remove the button below.

**Review by jonathanbardo — CHANGES_REQUESTED** [review]
This LGTM. Small changes on the LayoutPreview markup only.

File: src/extensions/layout-selector/layout-selector-results.js (hunk 4 above)
Comment (jrtashjian) [reply src/extensions/layout-selector/layout-selector-results.js:35]: Yeah, it was buried in the patterns api code. Works well :)

File: src/extensions/layout-selector/layout-selector-results.js (hunk 5 above)
Comment (jrtashjian) [reply src/extensions/layout-selector/layout-selector-results.js]: Resolved.
